app/controllers/UTRController.scala

```diff
@@ -105,17 +103,20 @@ class UTRController @Inject() (
               .bindFromRequest()
               .fold(
                 formWithErrors => render(mode, regime, formWithErrors, businessType).map(BadRequest(_)),
-                value =>
+                value => {
+                  val previousAnswers = request.userAnswers.get(UTRPage).contains(value)
                   (for {
-                    updatedAnswers <- setEither(UTRPage, value, checkPrevious = true)
+                    updatedAnswers <- request.userAnswers.set(UTRPage, value)
                     _ = sessionRepository.set(updatedAnswers)
                   } yield Redirect(navigator.nextPage(UTRPage, mode, regime, updatedAnswers)))
-                    .valueOrF(
+                    .fold(
                       _ =>
                         renderer
                           .render("thereIsAProblem.njk", Json.obj("regime" -> regime.toUpperCase, "emailAddress" -> appConfig.emailEnquiries))
-                          .map(ServiceUnavailable(_))
+                          .map(ServiceUnavailable(_)),
+                      result => Future.successful(result)
                     )
```

Comment:
this can be simplified to:

`value =>
                  for {
                    updatedAnswers <- Future.fromTry(request.userAnswers.set(UTRPage, value))
                    _              <- sessionRepository.set(updatedAnswers)
                  } yield Redirect(navigator.nextPage(UTRPage, mode, regime, updatedAnswers))`

and the errorHandler will take care of the failure case so there is no need for the fold

Reply:
Agree